backend/src/v5/models/revisions.js

```diff
@@ -42,43 +45,81 @@ const findOneRevisionByQuery = async (teamspace, model, query, projection, sort)
 Revisions.getLatestRevision = (teamspace, model, projection = {}) => {
 	const query = { ...excludeVoids, ...excludeIncomplete };
 	const sort = { timestamp: -1 };
-	return findOneRevisionByQuery(teamspace, model, query, projection, sort);
+	return findOneRevisionByQuery(teamspace, model, undefined, query, projection, sort);
 };
 
 Revisions.getRevisionCount = (teamspace, model) => {
 	const query = { ...excludeVoids, ...excludeIncomplete };
-	return db.count(teamspace, collectionName(model), query);
+	return db.count(teamspace, collectionName(undefined, model), query);
 };
 
-Revisions.getRevisions = (teamspace, model, showVoid, projection = {}) => {
+Revisions.getRevisions = (teamspace, model, modelType, showVoid, projection = {}) => {
 	const query = { ...excludeIncomplete };
 
 	if (!showVoid) {
 		query.void = excludeVoids.void;
 	}
 
-	return findRevisionsByQuery(teamspace, model, query, projection, { timestamp: -1 });
+	return findRevisionsByQuery(teamspace, model, modelType, query, projection, { timestamp: -1 });
 };
 
-Revisions.getRevisionByIdOrTag = (teamspace, model, revision, projection = {}) => findOneRevisionByQuery(teamspace,
-	model, { $or: [{ _id: revision }, { tag: revision }] }, projection);
+Revisions.getRevisionByIdOrTag = (teamspace, model, modelType, revision, projection = {}) => findOneRevisionByQuery(
+	teamspace, model, modelType, { $or: [{ _id: revision }, { tag: revision }] }, projection);
 
-Revisions.updateRevisionStatus = async (teamspace, project, model, revision, status) => {
-	const res = await db.findOneAndUpdate(teamspace, collectionName(model),
+Revisions.addRevision = async (teamspace, project, model, modelType, data) => {
+	const newRevision = {
+		_id: generateUUID(),
+		project,
+		model,
+		timestamp: new Date(),
+		...data,
+	};
+
+	await db.insertOne(teamspace, collectionName(modelType, model), newRevision);
+
+	publish(events.NEW_REVISION, { teamspace, project, model, modelType, revision: newRevision._id });
```

Comment:
This doesn't feel like is the right time for `NEW_REVISION`

This should only trigger when the processing has completed